android/src/main/kotlin/com/appcues/flutter/AppcuesFrameViewFactory.kt

```diff
@@ -0,0 +1,104 @@
+package com.appcues.flutter
+
+import android.content.Context
+import android.view.View
+import android.widget.FrameLayout
+import androidx.core.view.children
+import com.appcues.AppcuesFrameView
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.EventChannel
+import io.flutter.plugin.common.EventChannel.EventSink
+import io.flutter.plugin.common.StandardMessageCodec
+import io.flutter.plugin.platform.PlatformView
+import io.flutter.plugin.platform.PlatformViewFactory
+
+class AppcuesFrameViewFactory(
+    private val plugin: AppcuesFlutterPlugin,
+    private val messenger: BinaryMessenger,
+) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
+
+    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
+        val platformView = AppcuesPlatformView(context)
+
+        platformView.wrapperView.eventChannel =
+            EventChannel(messenger, "com.appcues.flutter/frame/$viewId")
```

Comment:
what is this? is it just a hardcoded channel name?

Reply:
The `$viewId` part is the key. Each native view that is created by the framework gets a unique ID, and we can use a distinct channel based on this ID to communicate the height updates back to the dart side, to update the `SizedBox` that is containing his native view.

That part happens here https://github.com/appcues/appcues-flutter-plugin/blob/77457d00c71ee5d3a4b8e470a1b2b91678a65d4d/lib/appcues_flutter.dart#L119-L125. Where the same `id` value is made available to the Dart callback, to tie them together.